app/nexus/internal/state/backend/sqlite/ddl/statements.go

```diff
@@ -30,23 +31,27 @@ CREATE TABLE IF NOT EXISTS secrets (
 CREATE TABLE IF NOT EXISTS secret_metadata (
 	path TEXT PRIMARY KEY,
 	current_version INTEGER NOT NULL,
+	oldest_version INTEGER NOT NULL,
 	created_time DATETIME NOT NULL,
-	updated_time DATETIME NOT NULL
+	updated_time DATETIME NOT NULL,
+	max_versions INTEGER NOT NULL
 );
 
 CREATE INDEX IF NOT EXISTS idx_secrets_path ON secrets(path);
 CREATE INDEX IF NOT EXISTS idx_secrets_created_time ON secrets(created_time);
 `
 
 // QueryUpdateSecretMetadata is a SQL query for inserting or updating secret
-// metadata. It updates the current version and updated time in conflict with
+// metadata. It updates the current version, oldest version, max versions, and updated time in conflict with
 // the existing path.
 const QueryUpdateSecretMetadata = `
-INSERT INTO secret_metadata (path, current_version, created_time, updated_time)
-VALUES (?, ?, ?, ?)
+INSERT INTO secret_metadata (path, current_version, oldest_version, created_time, updated_time, max_versions)
```

Comment:
version information was missing and that was causing test failures.

we didn't realize that before because things were cached in memory; but it was a bug for sure.